planet/src/main/webapp/js/authentication.js

```diff
@@ -40,9 +43,26 @@ document.addEventListener('DOMContentLoaded', function() {
     loadElement('signup.html', 'sign-up-modal');
 });
 
+function checkUserSignIn() {
+    currentUser = firebase.auth().currentUser;
```

Comment:
this is implicitly global. Declare it explicitly as a global for clarity

Reply:
Fixed!